### PR TITLE
Event results should maintain an overall colvis when groupby and redu…

### PR DESCRIPTION
**Pull request by jschmidt10**
We have a bug where using a group by query along with a reduced.response parameter results in the overall Event columnVisibility being empty.

The bug fix I did was the path of least risk/resistance. Now, with reduced.response, we will still serialize the field visibilities from the tserver back to the web tier. The transformer on the web side will eliminate them in the actual HTTP responses. 

There's an optimization where we change the Document to always serialize it's top level visibility so that we don't waste cycles serializing the individual field visibilities when using reduced.response. However, that'll take more work and testing. I can open up a follow on ticket to do that if we're interested. For now, this fixes the immediate issue.

### CODE REVIEW

**Review by apmoriarty — APPROVED** [review]

File: warehouse/query-core/src/main/java/datawave/query/attributes/Attribute.java [hunk 1]
```diff
@@ -155,64 +155,48 @@ protected void clearMetadata() {
...
-            out.writeBoolean(isMetadataSet());
-            if (isMetadataSet()) {
-                byte[] cvBytes = getColumnVisibility().getExpression();
-                
-                WritableUtils.writeVInt(out, cvBytes.length);
-                
-                out.write(cvBytes);
-                out.writeLong(getTimestamp());
-            }
+        out.writeBoolean(isMetadataSet());
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/attributes/Attribute.java:158]: This is no longer based on the reducedResponse boolean being sent in.  Is that what you meant to do?  I think the metadata will basically always be set on the attribute.
The reduced response was an effort to reduce the amount of data being sent back to the client (webserver/microservice).  If you are removing that concept altogether, then we shouldn't have a reduced response boolean here nor should that option be sent from the DefaultQueryPlanner over to the QueryIterator. via the scan options.

**Review by ivakegg — CHANGES_REQUESTED** [review]

File: warehouse/query-core/src/main/java/datawave/query/attributes/Attribute.java (hunk 1 above)
Comment (jschmidt10) [reply warehouse/query-core/src/main/java/datawave/query/attributes/Attribute.java:158]: Yes, that is what I meant to do. I can work on ripping it all the way out. I was initially targeting a quick bug fix to get the events marked correctly. I'll start working on fully taking it out though.

**Review by ivakegg — APPROVED** [review]

File: warehouse/query-core/src/main/java/datawave/query/attributes/Attribute.java (hunk 1 above)
Comment (jschmidt10) [reply warehouse/query-core/src/main/java/datawave/query/attributes/Attribute.java:158]: Had a discussion with Ivan. We're going to merge this as-is and I opened up a follow on ticket for a better long-term fix, which will be serializing the document level colvis always and then reverting this change.

**Comment by keith-ratcliffe** [comment]
The download.jboss.org endpoint for the Wildfly tarball was temporarily down, causing jobs to fail, but seems to be back online now. Rerunning jobs...

**Comment by keith-ratcliffe** [comment]
> The download.jboss.org endpoint for the Wildfly tarball was temporarily down, causing jobs to fail, but seems to be back online now. Rerunning jobs...

Scratch that....the URL needs to be changed to https:// in order for it to work. Update forthcoming

**Pull request merged**